packages/core/src/utils/requestQueue.ts

```diff
@@ -173,3 +257,28 @@ function shouldRetry(error: Error) {
   }
   return true;
 }
+
+export function resolveWebsocketTransport(
+  transport: ReturnType<Transport>,
+): ResolvedWebSocketTransport | undefined {
+  if (transport.config.type === "http") {
+    return undefined;
+  }
+
+  if (transport.config.type === "fallback") {
+    const fallbackTransport: ReturnType<FallbackTransport> =
+      transport as ReturnType<FallbackTransport>;
+
+    const wsTransport = fallbackTransport.value!.transports.find(
+      (t: ReturnType<Transport>) => t.config.type === "webSocket",
+    ) as ResolvedWebSocketTransport | undefined;
+
+    return wsTransport;
+  }
+
+  if (transport.config.type === "webSocket") {
+    return transport as ResolvedWebSocketTransport;
+  }
+
+  return undefined;
+}
```

Comment:
What do you think about getting away from using the `Transport` type in this module?

In the future this would let us implement features that aren't available in the transports like request cancellation. I also feel confident that `ponder.config.ts` should use rpc urls, not transports.

Reply:
I think this makes sense, and we can make this change once we get to config and switch to rpcUrls instead of transports.